src/transformers/models/fastspeech2/modeling_fastspeech2.py

```diff
@@ -0,0 +1,700 @@
+                ]
+                + ([nn.Tanh()] if i < n_layers - 1 else [])
+                + [nn.Dropout(dropout)]
+            )
+            nn.init.xavier_uniform_(
+                cur_layers[0].weight,
+                torch.nn.init.calculate_gain("tanh" if i < n_layers - 1 else "linear"),
+            )
+            self.convolutions.append(nn.Sequential(*cur_layers))
+
+    def forward(self, x):
+        x = x.transpose(1, 2)  # B x T x C -> B x C x T
+        for conv in self.convolutions:
+            x = conv(x)
+        return x.transpose(1, 2)
+
+
+class FastSpeech2Encoder(nn.Module):
+    def __init__(self, args):
+        super().__init__()
+        self.args = args
+        self.padding_idx = args.pad_token_id
+        self.n_frames_per_step = args.n_frames_per_step
+        self.out_dim = args.output_frame_dim * args.n_frames_per_step
+
+        self.spk_emb_proj = None
+        self.embed_speaker = None
+        if args.num_speakers > 1:
+            self.embed_speaker = nn.Embedding(args.num_speakers, args.speaker_embed_dim)
+            self.spk_emb_proj = nn.Linear(args.encoder_embed_dim + args.speaker_embed_dim, args.encoder_embed_dim)
+
+        self.dropout_module = nn.Dropout(args.dropout)
+        self.embed_tokens = nn.Embedding(args.vocab_size, args.encoder_embed_dim, padding_idx=self.padding_idx)
+
+        self.embed_positions = FastSpeech2PositionalEmbedding(
+            args.max_source_positions, args.encoder_embed_dim, self.padding_idx
+        )
+        self.pos_emb_alpha = nn.Parameter(torch.ones(1))
+        self.dec_pos_emb_alpha = nn.Parameter(torch.ones(1))
+
+        self.encoder_fft_layers = nn.ModuleList(
+            FFTLayer(
+                args.encoder_embed_dim,
+                args.encoder_attention_heads,
+                args.fft_hidden_dim,
+                args.fft_kernel_size,
+                dropout=args.dropout,
+                attention_dropout=args.attention_dropout,
+            )
+            for _ in range(args.encoder_layers)
+        )
+
+        self.var_adaptor = VarianceAdaptor(args)
+
+        self.decoder_fft_layers = nn.ModuleList(
+            FFTLayer(
+                args.decoder_embed_dim,
+                args.decoder_attention_heads,
+                args.fft_hidden_dim,
+                args.fft_kernel_size,
+                dropout=args.dropout,
+                attention_dropout=args.attention_dropout,
+            )
+            for _ in range(args.decoder_layers)
+        )
+
+        self.out_proj = nn.Linear(args.decoder_embed_dim, self.out_dim)
+
+        self.postnet = None
+        if args.add_postnet:
+            self.postnet = Postnet(
+                self.out_dim,
+                args.postnet_conv_dim,
+                args.postnet_conv_kernel_size,
+                args.postnet_layers,
+                args.postnet_dropout,
+            )
+
+        if args.mean:
+            self.register_buffer("mean", torch.zeros(self.out_dim))
```

Comment:
Mean and standard deviation are 0 and 1 by default so that even if the user forgets to use `set_mean` and `set_std`, the output is not affected.

Reply:
Wouldn't it be cleaner to let `mean` be of type `float` and set it to `0.0` by default (same with variance)? The `if config.mean()` is a bit weird to me

Reply:
I changed the flag name to `config.use_mean`. I also added an informative warning in case the user sets `use_mean` to `True`, pointing them to the `set_mean()` function (likewise for variance). 

Do you think it makes more sense for the user to specify the mean and variance explicitly in the config object? If so, one problem is that the mean and variance are 80-d vectors, so I'm afraid that it would pollute `config.json` too much. I might have misunderstood your intent though, definitely open to suggestions as I also think there might be a cleaner way to achieve this!